demo/src/main/java/com/example/EventHub/Event/EventService.java

```diff
@@ -140,4 +147,20 @@ public boolean errorEventStatus(EventDTO eventDTO) {
         }
         return false;
     }
+    public void apply(@RequestParam(name = "eventId") Integer id) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String username = authentication.getName();
+        User user = userRepository.findByFullName(username).get();
```

Comment:
You should add firstly Optional<User>optionalUser=userRepository.findByFullName(username)
if(optionalUser.isPresent()){
User user = optionalUser.get();
}